cardano-chain-gen/test/Test/Cardano/Db/Mock/Unit/Babbage/Tx.hs

```diff
@@ -59,3 +62,53 @@ consumeSameBlock =
     assertBlockNoBackoff dbSync 1
   where
     testLabel = "consumeSameBlock"
+
+isConsumeTxOutMigrated :: IOManager -> [(Text, Text)] -> Assertion
+isConsumeTxOutMigrated = do
+  let txOutParam =
+        TxOutParam
+          { paramMigrateConsumed = True,
+            paramPruneTxOut = False
+          }
+  withTxOutParamConfig txOutParam babbageConfigDir testLabel $ \interpreter mockServer dbSync -> do
+    void $ withBabbageFindLeaderAndSubmitTx interpreter mockServer $
+      Babbage.mkPaymentTx (UTxOIndex 0) (UTxOIndex 1) 10000 500
+
+    startDBSync dbSync
+    assertBlockNoBackoff dbSync 1
+    assertEqQuery dbSync DB.isMigrated True "missing consumed_by_tx_in_id column when flag --consumed-tx-out active"
+  where
+    testLabel = "isConsumeTxOutMigrated"
+
+pruneTxoutWhenFlagActive :: IOManager -> [(Text, Text)] -> Assertion
+pruneTxoutWhenFlagActive = do
+  let txOutParam =
+        TxOutParam
+          { paramMigrateConsumed = True,
+            paramPruneTxOut = True
+          }
+  withTxOutParamConfig txOutParam babbageConfigDir testLabel $ \interpreter mockServer dbSync -> do
+
+    startDBSync dbSync
+    -- fill in some blocks
+    b1 <- fillEpochPercentage interpreter mockServer 50
```

Comment:
forgeAndSubmitBlocks could be used instead to create a specific number of blocks.